homedir/homedir_test.go

```diff
@@ -0,0 +1,130 @@
+// https://github.com/mitchellh/go-homedir/blob/af06845cf3004701891bf4fdb884bfe4920b3727/homedir_test.go
+package homedir
+
+import (
+	"os"
+	"os/user"
+	"path/filepath"
+	"testing"
+)
+
+func patchEnv(key, value string) func() {
+	bck := os.Getenv(key)
+	deferFunc := func() {
+		os.Setenv(key, bck)
+	}
+
+	if value != "" {
+		os.Setenv(key, value)
+	} else {
+		os.Unsetenv(key)
+	}
+
+	return deferFunc
```

Comment:
oh thats a neat pattern. rather than doing a `WithThingFixed(f func)` and "wrapping" the func, just return the thing to be defer'd. 

course, then you might have to `defer` across multiple returns...